src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -874,6 +874,44 @@ string M(object o, bool b)
             return x;
         }
     }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        [WorkItem(25993, "https://github.com/dotnet/roslyn/issues/25993")]
+        public async Task TestEmbeddedStatement()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        var fe = e as C;
+        [|var|] ae = e as C;
+        if (fe != null)
+        {
+            M(fe);
+        }
+        else if (ae != null)
+        {
+            M(ae);
+        }
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        var fe = e as C;
+        if (fe != null)
+        {
+            M(fe);
+        }
+        else if (e is C ae)
+        {
+            M(ae);
+        }
+    }
 }");
```

Comment:
can we have more tests.  For example, a test where 'ae' is used in the first part of the if block?

Reply:
VB tests

Reply:
given the complexity of all the code below that is doing all the flow analysis and whatnot, i would like a *large* swath of tests that are trying to hit basically every branch in that code.   that way, if/when we touch that code, we should see breaks and we should be able to then tell precisely what sort of user code cases all that flow analysis code is trying to handle.  Thanks!

Reply:
>VB tests

Very Bad?

Reply:
oh.  nevermind.  forgot what feature hti was for a moment :D